### ISSUE TITLE
ADFS Module Issues

### ISSUE THREAD

**Issue by puzzlepeaches** [Tw1sm/spraycharles#14]
So normally I would fix this myself, but I can't for the life of me figure out how I would following the refactor so far. (Classes still confuse me sometimes!) When using the Adfs.py module, BaseHTTPTarget.py is throwing an error when trying to print the login attempt to console:

```python
        print(
            "%-35s %-17s %13s %15s"
            % (self.data["username"], self.data["password"], code, length)
        )
```

KeyError thrown is shown in the screenshot below:

<img width="1250" alt="image" src="https://user-images.githubusercontent.com/8538866/162182046-5e021f3d-bd79-459a-a459-45d2bdb8a1ac.png">

I think it may be due to the required casing for the username and password variables used in the ADFS module:

```python
        self.data = {
            "UserName": "",
            "Password": "",
            "AuthMethod": "FormsAuthentication",
        }

    def set_username(self, username):
        self.data["UserName"] = username

    def set_password(self, password):
        self.data["Password"] = password
```

If I were to modify the the username and password variable in the code snippet above, the module runs but the app throws 500 responses instead of the expected 200.

Looking at browser login requests to a valid ADFS portal shows that the UserName/Password casing is required for successful login so we can't simply modify the POST variables:

<img width="346" alt="image" src="https://user-images.githubusercontent.com/8538866/162182700-f3fed163-f3c5-4495-81eb-a3e1180b0444.png">

Let me know your thoughts. Thanks!

**Comment by Tw1sm** [comment]
Try checking out #15 and let me know if it solves the issue. Didn't have ability to test it much so it may not be a complete fix yet.

Good catch here, I clearly didn't test the parent class refactor enough.

**Comment by puzzlepeaches** [comment]
Looks like the change is working! Should be able to close this out after following a merge of #15.

**Comment by Tw1sm** [comment]
Fixed with de0850409fcbc850210c1df4004591623bcfb9ab. Could still be improved to remove much of the code duplication with the `set_username()` and `set_password()` methods on targets inheriting from `BaseHttpTarget`, but felt the need to get the fix into master sooner rather than later.

Thanks for the screenshots and documentation on the issue!
